stdlib/Distributed/src/remotecall.jl

```diff
@@ -72,7 +72,7 @@ function test_existing_ref(r::AbstractRemoteRef)
         if isa(r, Future) && found.v === nothing && r.v !== nothing
```

Comment:
it looks like this function may require the `client_refs` lock, since their are multiple uses of that dict

Reply:
getkey is locking on client_refs, but still that function looks fishy
In the end i didn't find any issues with it, but I'll keep an eye on it

Reply:
It is generally insufficient to lock each use independently, but all uses generally must share the same lock